atomate/atat/fireworks/core.py

```diff
@@ -0,0 +1,193 @@
+        1 minute will be deducted from this to run the database
+        tasks.
+        :param: ncores (int): number of instances of mcsqs
+        to run (mcsqs is not parallelized, these are run
+        independently of each other), by default will try
+        to detect automatically
+        :param: db_file:
+        :param kwargs: Other kwargs that are passed to Firework.__init__.
+        """
+
+        if walltime < 3:
+            raise ValueError("Please increase walltime.")
+
+        if disordered_struct.is_ordered:
+            raise ValueError("You must input a disordered structure.")
+
+        if db_file is None:
+            logger.warn("Database not specified for McsqsFW.")
+
+        # preprocess disordered structure (scales occupancies, normalizes volume)
+        disordered_struct = preprocess_disordered_structure(disordered_struct,
+                                                            max_denominator=max_denominator)
+
+        if size is None:
+            size = self._determine_min_cell(disordered_struct)*2
+        else:
+            size = self._determine_min_cell(disordered_struct)*size
+
+        if clusters is None:
+            # TODO: make cluster determination a bit smarter!
+            lattice_param = min(disordered_struct.lattice.abc)
+            clusters = {
+                -2: lattice_param*2.001,
+                -3: lattice_param*2.001,
+                -4: lattice_param*1.501
+            }
+        else:
+            for cluster in clusters.keys():
+                if cluster not in [2, 3, 4, 5, 6]:
+                    raise ValueError("Mcsqs only supports clusters of 2-6 atoms.")
+
+        rndstr = disordered_struct.to(fmt='mcsqs')
+
+        cluster_str = " ".join(["{}={}".format(k, v) for k, v in clusters.items()])
+        generate_cluster_cmd = "mcsqs {}".format(cluster_str)
+
+        if user_input_settings:
+            user_input_settings_str = " ".join(["{}={}".format(k, v) for k, v
+                                            in user_input_settings.items()])
+        else:
+            user_input_settings_str = ""
+
+        # using same approach as Custodian to detect ncores
+        ncores = os.environ.get('NSLOTS') or multiprocessing.cpu_count()
+        ncores = int(ncores)
+
+        # mcsqs is not parallelised, and is not expected to finish within the walltime
+        # it's a Monte Carlo process that can run indefinitely, so we use timeout
+        # to ensure the command finishes
+        timeout_cmd = which('timeout') or which('gtimeout')
+        if not timeout_cmd:
+            raise RuntimeError("Requires timeout to limit walltime.")
+
+        with open(os.path.join(module_dir, 'run_mcsqs_template.sh')) as f:
+            template = f.read()
+
+        bash_script = Template(template)
+        bash_script = bash_script.substitute({
+            'timeout_cmd': timeout_cmd,
+            'time': 60*(walltime - 1),  # seconds, minus 60 seconds for DB task
```

Comment:
I think you should rephrase the walltime parameter to only be the SQS time. In the docs for walltime just say that one should leave at least 1 minute for the db insertion.